### ISSUE TITLE
EEG-TMS Processing Example in MNE

### ISSUE THREAD

**Issue by padmasundaram** [mne-tools/mne-python#7009]
Hi,

@Davi1990 and I would like to create an EEG-TMS example based on the EEG-TMS dataset provided in FieldTrip. @jasmainak suggested that we would need to get permission to fetch the data and use it in the example. There is interest in having a EEG-TMS example in MNE as several groups are trying to do similar preprocessing and analysis. So it would be good to have a example pipeline.

This is the FieldTrip example I am referring to:
http://www.fieldtriptoolbox.org/tutorial/tms-eeg/#dataset-information

**Comment by agramfort** [comment]
@robertoostenveld is sitting next to me. He is fine with it but we need to make sure we give credit to fieldtrip and authors of the dataset (references etc). So +1

**Comment by padmasundaram** [comment]
Thanks @robertoostenveld! Of course, we will credit field trip and the dataset authors/creators. 
I don't see a structural MRI for that dataset. Is it just an EEG data? I would like to have the full pipeline in the example with the inverse solution plotted on the cortex.

**Comment by robertoostenveld** [comment]
The TMS-EEG tutorial does not include source reconstruction, it aims at cleaning and some simple channel level analyses. Afaik we also don't have the anatomical MRI of this subject. See http://www.fieldtriptoolbox.org/tutorial/tms-eeg/

**Comment by padmasundaram** [comment]
We will have to use our own data for the source reconstruction example in that case.

**Comment by agramfort** [comment]
would there be something specific about EEG+TMS for source imaging? Or once
I have cleaned up the sensor data I can just use any method I know? what
would we willing to demonstrate?

>

**Comment by JonteP** [comment]
Did anything ever come of this? I would be very interested in such an example pipeline!

**Comment by padmasundaram** [comment]
@JonteP : Is it enough to just remove artifacts from the EEG like in the FieldTrip example? Do you have a dataset in mind or do you want to just use FieldTrip's example?

**Comment by JonteP** [comment]
Personally, I'm looking for an alternative to the TESA toolbox and pipeline
(https://nigelrogasch.gitbook.io/tesa-user-manual/example_pipelines) since
I'm migrating from Matlab to Python. But I am interested in anything to get
me started and I guess it needs to be general enough to serve as a general
TMS-EEG tutorial.

:)

**Comment by jasmainak** [comment]
I agree that TESA is a high priority ... it's something many folks have asked for and certain parts are not too difficult to implement

**Comment by padmasundaram** [comment]
I am open to working together on this. I would use the TMS-EEG application for my own project. Can we put this on a list for a code sprint?

**Comment by Davi1990** [comment]
Hi @padmasundaram and @jasmainak! I am down to contribute. For preprocessing I use MATLAB and EEGLAB but I would be more happy to do all of the analyses in MNE tbh.

**Comment by JonteP** [comment]
Glad to see there is interest in this! I cannot contribute much, except perhaps example data with our somewhat unusual equipment, if ever needed.
